File: easy/query_board.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strings"
)

func main() {
	var p, v int
	data, err := os.Open(os.Args[1])
	if err != nil {
		log.Fatal(err)
	}
	defer data.Close()
	m := make([][]int, 256)
	for i := 0; i < 256; i++ {
		m[i] = make([]int, 256)
	}
	scanner := bufio.NewScanner(data)
	for scanner.Scan() {
		t := strings.Fields(scanner.Text())
		if strings.HasPrefix(t[0], "Query") {
			var sum int
			fmt.Sscan(t[1], &p)
			if strings.HasSuffix(t[0], "Row") {
				for i := 0; i < 256; i++ {
					sum += m[p][i]
				}
			} else {
				for i := 0; i < 256; i++ {
					sum += m[i][p]
				}
			}
			fmt.Println(sum)
		} else {
			fmt.Sscan(t[1], &p)
			fmt.Sscan(t[2], &v)
			if strings.HasSuffix(t[0], "Row") {
				for i := 0; i < 256; i++ {
					m[p][i] = v
				}
			} else {
				for i := 0; i < 256; i++ {
					m[i][p] = v
				}
			}
		}
	}
}

File: easy/query_board.lua
m, n = {}, 256
for line in io.lines(arg[1]) do
  if line:find("Query") then
    local q, sum = line:match(" (%d+)"), 0
    if line:find("Row") then
      for i = 1, n do
        if m[(q-1)*n + i] then
          sum = sum + m[(q-1)*n + i]
        end
      end
    else
      for i = 1, n do
        if m[(i-1)*n + q] then
          sum = sum + m[(i-1)*n + q]
        end
      end
    end
    print(sum)
  else
    local p, v = line:match(" (%d+) (%d+)")
    if line:find("Row") then
      for i = 1, n do m[(p-1)*n + i] = v end
    else
      for i = 1, n do m[(i-1)*n + p] = v end
    end
  end
end
